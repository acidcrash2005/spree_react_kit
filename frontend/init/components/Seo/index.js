import React, { Component } from 'react';
import { Helmet } from "react-helmet";

export default class Seo extends Component {
    render () {
        const { title } = this.props;

        return (
            <Helmet>
                <title>{title}</title>
            </Helmet>
        );
    }
}
