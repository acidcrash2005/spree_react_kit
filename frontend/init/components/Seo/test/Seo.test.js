// Core
import React from 'react';
import { mount } from 'enzyme';

// Test component
import Seo from '../';

const props = {
    title: 'TestString',
};

const result = mount(<Seo { ...props } />);

describe('Компонент Seo:', () => {
    describe('должен иметь базовую разметку JSX:', () => {
        test('должен соответствовать снепшоту разметки', () => {
            expect(result).toMatchSnapshot();
        });
    });

    describe('должен иметь базовые пропсы, полученные от родительского компонента:', () => {
        describe('title — заголовок для <title>', () => {
            test('должен существовать', () => {
                expect(result.prop('title')).toBeDefined();
            });
            test('должен иметь строковый тип', () => {
                expect(typeof result.prop('title')).toBe('string');
            });
        });
    });
});
