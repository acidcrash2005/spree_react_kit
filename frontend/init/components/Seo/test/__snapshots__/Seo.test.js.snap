// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Компонент Seo: должен иметь базовую разметку JSX: должен соответствовать снепшоту разметки 1`] = `
<Seo
  title="TestString"
>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <SideEffect(NullComponent)
      defer={true}
      encodeSpecialCharacters={true}
      title="TestString"
      titleAttributes={Object {}}
    >
      <NullComponent
        defer={true}
        encodeSpecialCharacters={true}
        title="TestString"
        titleAttributes={Object {}}
      />
    </SideEffect(NullComponent)>
  </HelmetWrapper>
</Seo>
`;
